src/gui/qgsmapcanvas.cpp

```diff
@@ -1023,14 +1023,15 @@ void QgsMapCanvas::zoomToSelected( QgsVectorLayer *layer )
   }
 
   rect = mapSettings().layerExtentToOutputExtent( layer, rect );
+  QgsPointXY center = rect.center();
```

Comment:
Shouldn't this stay inside the if clause on line 1030?
That looks like an unnecessary calculation in other cases.

Reply:
You are correct.